src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -604,44 +604,119 @@ internal BoundStatement BindLocalDeclarationStatement(LocalDeclarationStatementS
 
         private BoundStatement BindDeclarationStatementParts(LocalDeclarationStatementSyntax node, DiagnosticBag diagnostics)
         {
-            var typeSyntax = node.Declaration.Type.SkipRef(out _);
+            
             bool isConst = node.IsConst;
-
+            var typeSyntax = node.Declaration.Type.SkipRef(out _);
             bool isVar;
             AliasSymbol alias;
             TypeSymbol declType = BindVariableType(node.Declaration, diagnostics, typeSyntax, ref isConst, isVar: out isVar, alias: out alias);
 
-            // UNDONE: "possible expression" feature for IDE
-
-            LocalDeclarationKind kind = LocalDeclarationKind.RegularVariable;
-            if (isConst)
+            if (node.UsingKeyword != default)
             {
-                kind = LocalDeclarationKind.Constant;
+                Conversion iDisposableConversion;
+                MethodSymbol disposeMethod;
+                var declarations = BindUsingVariableDeclaration(
+                                                    this,
+                                                    diagnostics,
+                                                    diagnostics.HasAnyErrors(),
+                                                    node,
+                                                    node.Declaration,
+                                                    out iDisposableConversion,
+                                                    out disposeMethod);
+                return new BoundUsingLocalDeclarations(node, disposeMethod, iDisposableConversion, declarations);
             }
-            else if (node.UsingKeyword != default)
+            else
             {
-                kind = LocalDeclarationKind.UsingVariable;
+                LocalDeclarationKind kind = LocalDeclarationKind.RegularVariable;
```

Comment:
You could probably write this

`var kind = isConst ? LocalDeclarationKind.Constant : LocalDeclarationKind.RegularVariable;`